File: Task2/swift/main.swift
import Foundation

// Чтение количества колец
print("Количество колец: ")
guard let nStr = readLine(), let n = Int(nStr) else {
    print("Ошибка ввода количества колец")
    exit(1)
}
print("Строка с кольцами: ")
guard let ringsStr = readLine() else {
    print("Ошибка ввода строки колец")
    exit(1)
}

let rings = Array(ringsStr)

if rings.count < n * 2 {
    print("Строка с кольцами слишком короткая")
    exit(1)
}

// Массив для хранения множеств цветов на 10 стержнях
var colors = Array(repeating: Set<Character>(), count: 10)

// Обработка каждого кольца
for i in 0..<n {
    let color = rings[i*2]
    let rodChar = rings[i*2 + 1]

    guard let rod = Int(String(rodChar)), rod >= 0, rod <= 9 else {
        print("Ошибка: неверный номер стержня")
        continue
    }

    colors[rod].insert(color)
}

// Подсчёт стержней с кольцами всех цветов
var result = [Int]()
for i in 0..<10 {
    if colors[i].contains("r") && colors[i].contains("g") && colors[i].contains("b") {
        result.append(i)
    }
}

print("Количество стержней с кольцами всех трёх цветов: \(result.count)")
if result.isEmpty {
    print("Нет стержней с тремя цветами")
} else {
    print("Номера этих стержней: ", terminator:"")
    for r in result {
        print("\(r) ", terminator:"")
    }
    print()
}

File: Task2/javaScript/main.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Запрашиваем количество колец
rl.question("Введите количество колец: ", (nInput) => {
    const n = parseInt(nInput);

    // Проверка корректности числа
    if (isNaN(n) || n <= 0) {
        console.log("Ошибка: введите корректное положительное число.");
        rl.close();
        return;
    }

    // Запрашиваем строку с кольцами
    rl.question("Введите строку с кольцами: ", (ringsStr) => {

        // Проверка длины строки
        if (ringsStr.length !== n * 2) {
            console.log("Ошибка в длине строки");
            rl.close();
            return;
        }

        // Массив множеств цветов для 10 стержней
        const colors = Array.from({ length: 10 }, () => new Set());

        // Обработка колец
        for (let i = 0; i < n; i++) {
            const color = ringsStr[i * 2];
            const rodChar = ringsStr[i * 2 + 1];

            // Проверка цвета
            if (!['r', 'g', 'b'].includes(color)) {
                console.log(`Ошибка в воде строки`);
                rl.close();
                return;
            }

            // Проверка стержня
            const rod = parseInt(rodChar);
            if (isNaN(rod) || rod < 0 || rod > 9) {
                console.log(`Ошибка в воде строки`);
                rl.close();
                return;
            }

            colors[rod].add(color);
        }

        // Поиск стержней с тремя цветами
        const result = [];
        for (let i = 0; i < 10; i++) {
            if (colors[i].has('r') && colors[i].has('g') && colors[i].has('b')) {
                result.push(i);
            }
        }

        // Вывод результата
        console.log("Количество стержней с кольцами всех трёх цветов:", result.length);

        if (result.length === 0) {
            console.log("Нет стержней с тремя цветами");
        } else {
            console.log("Номера этих стержней:", result.join(' '));
        }

        rl.close();
    });
});
